src/dash-table/derived/table/fragments.tsx

```diff
@@ -19,6 +20,26 @@ function renderFragment(cells: any[][] | null, offset: number = 0) {
         null;
 }
 
+const getHiddenCell = (cell: JSX.Element) => React.cloneElement(cell, {
+    ...cell.props,
+    className: cell.props.className ? `${cell.props.className} phantom-cell` : 'phantom-cell'
+}, cell.type !== 'th' && cell.type !== 'td' ? cell.props.children : null);
+
+const getFixedColSpan = (cell: JSX.Element, maxColSpan: number) => React.cloneElement(cell, {
+    ...cell.props,
+    colSpan: R.isNil(cell.props.colSpan) ? cell.props.colSpan : Math.min(cell.props.colSpan, maxColSpan)
+});
+
+const getLastOfType = (cell: JSX.Element) => React.cloneElement(cell, {
+    ...cell.props,
+    className: cell.props.className ? `${cell.props.className} last-of-type` : 'last-of-type'
+});
+
+const getFirstPhantom = (cell: JSX.Element) => React.cloneElement(cell, {
+    ...cell.props,
+    style: { ...cell.props.style, borderLeft: 'transparent' }
+});
```

Comment:
Various component cloning / manipulations.

`getHiddenCell` makes a phantom version of a cell - will be invisible
`getFirstPhantom` solves a doubled border issue